python/tank/platform/engine.py

```diff
@@ -1504,6 +1550,11 @@ def start_engine(engine_name, tk, context):
     # register this engine as the current engine
     set_current_engine(obj)
 
+    # ensure the metrics queue is initialized (dispatchers are running)
+    metrics_queue = MetricsQueue()
+    if not metrics_queue.initialized:
+        metrics_queue.init(tk)
```

Comment:
I was picturing something like this down in the API instance rather than up at the engine level.  There is the possibility of logging things like commands being run, which would need this all at a lower level, right?